### PR TITLE
refactor: scripts/VISS_Sample_Data.R

### PR DESCRIPTION
**Pull request by IshikaKhandelwal**
This pull request includes several significant updates to the project, primarily focusing on importing new dependencies, adding new functions for climate data analysis, and updating the documentation. Below is a summary of the most important changes:

### Dependency Updates:
* Added new dependencies in the `DESCRIPTION` file: `dplyr`, `purrr`, `sf`, `tibble`, `terra`, `exactextractr`, `future`, and `furrr`.
* Updated the `NAMESPACE` file to import functions from the new dependencies.

### Documentation Updates:
* Added documentation for the new functions: `exposure`, `exposure_times`, `extract_climate_data`, `get_niche_limits`, and `prepare_range`. [[1]](diffhunk://#diff-1707627f913e68c82f153b2db7039891a27f30b376c08f6116f6fc63c1281101R1-R23) [[2]](diffhunk://#diff-33414ebd21bf3d3fc058cb8c3d3acae4097cb205d2ce7e6493325336e2146347R1-R21) [[3]](diffhunk://#diff-8b4d0cf355f0ccdb06cedfadbd392e4878213ea4656d20767dc74967feec790bR1-R19) [[4]](diffhunk://#diff-1d11cc1cbd1cd225e90b6683645934299670db03c4d10ae52e9a17378fa0bf2fR1-R19) [[5]](diffhunk://#diff-e81b6af22ccfdda3d44ea6c61538f96f0fa0efd334d561c1a4e45e24d639aab7R1-R19)

### Miscellaneous:
* Added global variables declaration to `R/globals.R` to avoid R CMD check warnings.- Modularized key functions (`prepare_range, extract_climate_data, get_niche_limits, exposure, exposure_times`) and moved them to the `/R` folder for better code reuse and maintainability.

- Created a comprehensive driver script `scripts/VISS_Sample_Data.R `to orchestrate the workflow, including range preparation, climate data extraction, thermal niche limit computation, exposure calculation, and exposure times analysis.

Addressing: https://github.com/uw-ssec/biodiversity-horizons/issues/22

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -1,4 +1,25 @@
 # Generated by roxygen2: do not edit by hand
 
 export("%>%")
+importFrom(dplyr,across)
```
Comment (carlosgjs) [inline NAMESPACE:4]: shouldn't we be also exporting the main package functions here? E.g. `exposure`, `exposure_times`, etc?

File: R/exposure_times.R [hunk 2]
```diff
@@ -0,0 +1,68 @@
+#' Calculate exposure times for each species
+#'
+#' @param data A row of exposure data
+#' @param original.state Initial exposure state
+#' @param consecutive.elements Minimum consecutive years for state change
+#' @return A tibble with exposure and de-exposure times
+#' @importFrom dplyr filter pull
+#' @importFrom tibble tibble
+exposure_times <- function(data, original.state, consecutive.elements) {
```
Comment (carlosgjs) [inline R/exposure_times.R:9]: Can you look into all these warnings?

**Review by carlosgjs — CHANGES_REQUESTED** [review]
this looks really good, mostly need to address the warnings and I have the question about exporting the core functions.

**Comment by IshikaKhandelwal** [comment]
I am currently facing an issue where no linting warnings appear when I test the package locally using `lintr::lint_package()` and after applying fixes like `usethis::use_import_from()`, adding proper imports in @importFrom directives, and ensuring the namespace is updated.

However, when I push the changes to GitHub, the CI workflow still reports linting warnings, particularly related to missing global function definitions (e.g., %>%, tibble, pull). Despite matching the local linter configuration to the one used in GitHub Actions, I am unable to reproduce these warnings locally, making it challenging to debug effectively.

Any insights on what might be causing this discrepancy or suggestions on how to proceed would be greatly appreciated!

Thanks!

**Comment by IshikaKhandelwal** [comment]
Added issue for tracking the warnings issue here: https://github.com/uw-ssec/biodiversity-horizons/issues/38

**Review by carlosgjs — APPROVED** [review]
The warnings will be addressed in https://github.com/uw-ssec/biodiversity-horizons/issues/38

**Comment by atambay37** [comment]
Closes #53 and #56

**Pull request merged**